core/modules/qdisp/Executive.h

```diff
@@ -61,7 +63,7 @@ class QueryResource;
 class Executive : public std::enable_shared_from_this<Executive> {
 public:
     typedef std::shared_ptr<Executive> Ptr;
-    typedef std::map<int, std::shared_ptr<JobQuery>> JobMap;
+    typedef std::unordered_map<int, std::shared_ptr<JobQuery>> JobMap;
```

Comment:
I'm not sure about scaling part, I saw cases when hash table was slower than tree, would be interesting to compare two in this case.